src/target/codegen.cc

```diff
@@ -47,7 +47,12 @@ runtime::Module Build(IRModule mod, const Target& target) {
           .value()) {
     mod = tir::transform::SkipAssert()(mod);
   }
-  std::string build_f_name = "target.build." + target->target_name;
+  std::string build_f_name;
+  if (target->id->name == "micro_dev") {
```

Comment:
This is a bit ad-hoc here. Later maybe we should register codegen entry function as an attribute attached to target_id

Reply:
in the future i'd like to remove `micro_dev` as a special target name, but we haven't come to consensus on that, so it's fine for now.

Reply:
@areusch yeah, this PR makes it happen. The `micro_dev` is promoted as a new target name, with "codegen_c" as its code generator.